### PR TITLE
Run Mode Selection via Command Line Arguments

### PR DESCRIPTION
**Pull request by mbolt35**
## What does this PR change?
* Introduces a single entry point that allows a command line selection of which mode to run cost-model in.
  * `cost-model`(default): Providing no arguments or `cost-model` will run the application exactly the same as before this change. 
  * `agent`: Using the `agent` command will run the cost-model in metrics-exporter/agent mode.

## How was this PR tested?
* Tested in an existing cluster without setting any command line argument (backwards compatibility testing)
* Tested in an existing cluster with `cost-model` argument passed in the pod spec (explicit argument test)
* Tested in an existing agent cluster with the `agent` argument passed in.

### CODE REVIEW

File: pkg/cmd/commands.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		Use:   CommandCostModel,
+		Short: "Cost-Model metric exporter and data aggregator.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return costmodel.Execute(opts)
+		},
+	}
+
+	// TODO: We could introduce a way of mapping input command-line parameters to a configuration
+	// TODO: object, and pass that object to the agent Execute()
+	// cmCmd.Flags().<Type>VarP(&opts.<Property>, "<flag>", "<short>", <default>, "<usage>")
```
Comment (michaelmdresser) [inline pkg/cmd/commands.go:88]: There's a pretty neat CLI library for Go called [kong](https://github.com/alecthomas/kong) that lets you do this with struct tags, might be worth a look!

File: cmd/costmodel/main.go [hunk 2]
```diff
@@ -1,29 +1,22 @@
...
-func Healthz(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
-	w.WriteHeader(200)
-	w.Header().Set("Content-Length", "0")
-	w.Header().Set("Content-Type", "text/plain")
-}
-
 func main() {
-	a := costmodel.Initialize()
+	klog.InitFlags(nil)
+	flag.Set("v", "3")
```
Comment (michaelmdresser) [inline cmd/costmodel/main.go:13]: I know this is pulled from legacy, but I'm curious if hardcoding the log level is really a good idea.

**Review by michaelmdresser — APPROVED** [review]
Looks great! One idea about an alternative CLI library for consideration's sake, but I don't have any major problems with cobra!

The `go.sum` got a lot of additional deps and versions, is that because of cobra?

File: cmd/costmodel/main.go (hunk 2 above)
Comment (mbolt35) [reply cmd/costmodel/main.go:13]: Yeah, I really like the simplicity of the klog library, but there are some weird things happening with this. I'm not sure if passing `-v` to the application overrides this or not. If that's the case, I'm fine with leaving the default at 3 (it looks like the default inside klog is 0).

**Pull request merged**